sencha-workspace/packages/slate-cbl/src/view/student/DashboardController.js

```diff
@@ -0,0 +1,137 @@
+            contentArea = siteEnv.cblContentArea && Ext.create('Slate.cbl.model.ContentArea', siteEnv.cblContentArea),
+            competenciesStore = Ext.getStore('cbl-competencies-loaded'),
+            competenciesTpl = Ext.XTemplate.getTpl(me.view, 'competenciesTpl');
+
+        if (!student || !contentArea) {
+            return;
+        }
+
+        // empty competencies list
+        studentDashboardCompetenciesList.empty();
+        studentDashboardCompetenciesList.removeCls('competencies-unloaded').addCls('competencies-loading');
+
+        contentArea.getCompetenciesForStudents([student.getId()], function(competencies) {
+            var competenciesLength = competencies.length,
+                competencyIndex = 0,
+                competency, skillsList;
+            
+
+            if(!competenciesStore.isLoaded()) {
+                competenciesStore.loadRawData(competencies);
+            }
+
+            competenciesTpl.overwrite(studentDashboardCompetenciesList, {
+                student: student.getData(),
+                competencies: competencies
+            });
+
+            studentDashboardCompetenciesList.removeCls('competencies-loading').addCls('competencies-loaded');
+
+            for (; competencyIndex < competenciesLength; competencyIndex++) {
+                competency = Ext.create('Slate.cbl.model.Competency', competencies[competencyIndex]);
+                skillsList = studentDashboardCompetenciesList.down('.cbl-competency-panel[data-competency="'+competency.getId()+'"] .cbl-skill-meter');     
+                me.loadSkills(student, competency, skillsList);
+            }
+        });
+    },
+    
+    onDemoCellClick: function(dashboardView, ev, targetEl) {
+
+        Ext.create('Slate.cbl.view.student.skill.OverviewWindow', {
+            autoShow: true,
+            animateTarget: targetEl,
+
+            student: window.SiteEnvironment.cblStudent.ID,
+            competency: targetEl.up('ul.cbl-skill-demos').up('li.cbl-competency-panel').getAttribute('data-competency'),
+            skill: targetEl.getAttribute('data-skill')
+        });
+    },
+    
+
+
+    // protected methods
+    loadSkills: function(student, competency, skillsList) {
+        var me = this,
+            skillsTpl = Ext.XTemplate.getTpl(me.view, 'skillsTpl'),
+            skills, demonstrations, _renderSkills;
+
+        skillsList.removeCls('skills-unloaded').addCls('skills-loading');
+
+        _renderSkills = function() {
+            var demonstrationsLength = demonstrations.length, demonstrationIndex = 0, demonstration, skill;
+            // group skills by student
+            for (; demonstrationIndex < demonstrationsLength; demonstrationIndex++) {
+                demonstration = demonstrations[demonstrationIndex];
+                skill = skills.get(demonstration.SkillID);
+
+                if (!skill.demonstrations) {
```

Comment:
I disagree with having superfluous typeof and null comparisons when the application doesn't strictly call for it, it adds bytes for no tangible reason and the added strictness actually prevents the code from running when it should. The simple falsey check is readable and behaves correctly.

`skills` is an instance of `Ext.util.Collection` and its get method will return null if the supplied key was not found. However in this particular case that is impossible, as skills is populated by a remote call to get all skills for a given compentencyId, and demonstrations is populated by a call to get all demonstrations for all the skills in a given competencyId. The only way this could fail is if the competency map was updated and the database made inconsistent between the two calls, and I think that's an exceptional enough case not to bloat the code handling considering the limited consequence.